tests/integration/rdb.tcl

```diff
@@ -198,3 +198,77 @@ test {client freed during loading} {
         exec kill [srv 0 pid]
     }
 }
+
+start_server {overrides {save ""}} {
+    test {Test child sending COW info} {
+        # make sure that rdb_last_cow_size is zero (the test using new server), so that the
+        # comparison at the end of the test will be valid
+        assert {[s rdb_last_cow_size] == 0}
+
+        # using a 200us delay, the bgsave is empirically taking about 10 seconds.
+        # we need it to take more than some 5 seconds, since redis only report COW once a second.
+        r config set rdb-key-save-delay 200
+
+        # populate the db with 10k keys of 4k each
+        set rd [redis_deferring_client 0]
+        set size 4096
+        set cmd_count 10000
+        for {set k 0} {$k < $cmd_count} {incr k} {
+            $rd set key$k [string repeat A $size]
+        }
+
+        for {set k 0} {$k < $cmd_count} {incr k} {
+            catch { $rd read }
+        }
+
+        $rd close
+
+        # start background rdb save
+        r bgsave
+
+        # on each iteration, we will write some key to the server to trigger copy-on-write, and
+        # wait to see that it reflected in INFO.
+        set iteration 1
+        while 1 {
+            # take a sample before writing new data to the server
+            set cow_info [s rdb_active_cow_size]
+            if {$::verbose} {
+                puts "COW info before copy-on-write: $cow_info"
+            }
+
+            # trigger copy-on-write
+            r setrange key$iteration 0 [string repeat B $size]
+
+            # wait to see that rdb_active_cow_size value updated (as long as the child is in progress)
+            wait_for_condition 80 100 {
+                [s rdb_bgsave_in_progress] == 0 ||
+                [s rdb_active_cow_size] >= $cow_info + $size
+            } else {
+                if {$::verbose} {
+                    puts "COW info on fail: [s rdb_active_cow_size]"
+                }
+                fail "COW info didn't report"
+            }
+
+            # for no accurate, stop after 2 iterations
+            if {!$::accurate && $iteration == 2} {
+                break
+            }
+
+            # stop iterating if the bgsave completed
+            if { [s rdb_bgsave_in_progress] == 0 } {
+                break
+            }
+
+            incr iteration 1
+        }
+
+        # if bgsave completed, check that rdb_last_cow_size value is at least as last rdb_active_cow_size.
+        if { [s rdb_bgsave_in_progress] == 0 } {
+            set last_cow_size [s rdb_last_cow_size]
+            if {$last_cow_size < $cow_info} {
+                fail "last COW size:$last_cow_size is less than $cow_info"
+            }
```

Comment:
this can be changed to ``assert { $a > $b}` or better yet `assert_lessthan $a $b`
in this case both will print the value and the condition when the assertion fails.
if it were `assert {$a < [s bbb]}` then the value of the `[]` would not be printed, which is why using `assert_lessthan` or `assert_equals` is better than just `assert`